src/service/action-impl.js

```diff
@@ -304,61 +311,267 @@ export class ActionService {
     if (actionMap === undefined) {
       actionMap = null;
       if (node.hasAttribute('on')) {
-        actionMap = this.parseActionMap_(node.getAttribute('on'));
+        actionMap = parseActionMap(node.getAttribute('on'), node);
       }
       node[ACTION_MAP_] = actionMap;
     }
     return actionMap;
   }
+}
 
-  /**
-   * @param {string} s
-   * @return {?Object<string, ActionInfoDef>}
-   */
-  parseActionMap_(s) {
-    let actionMap = null;
-    const actions = s.split(';');
-    if (actions && actions.length > 0) {
-      for (let i = 0; i < actions.length; i++) {
-        const actionStr = actions[i];
-        const actionInfo = this.parseAction_(actionStr);
-        if (actionInfo) {
-          if (!actionMap) {
-            actionMap = {};
-          }
-          actionMap[actionInfo.event] = actionInfo;
+
+/**
+ * @param {string} s
+ * @param {!Element} context
+ * @return {?Object<string, ActionInfoDef>}
+ * @private Visible for testing only.
+ */
+export function parseActionMap(s, context) {
+  function assertAction(condition, opt_message) {
+    return assert(condition, 'Invalid action definition in %s: [%s] %s',
+        context, s, opt_message || '');
+  }
+  function assertToken(tok, type, opt_value) {
+    if (opt_value !== undefined) {
+      assertAction(tok.type == type && tok.value == opt_value,
+          `; expected [${opt_value}]`);
+    } else {
+      assertAction(tok.type == type);
+    }
+    return tok;
+  }
+
+  let actionMap = null;
+
+  const toks = new ParserTokenizer(s);
+  let tok;
+  let peek;
+  do {
+    tok = toks.next();
+    if (tok.type == TokenType.EOF ||
+            tok.type == TokenType.SEPARATOR && tok.value == ';') {
+      // Expected, ignore.
+    } else if (tok.type == TokenType.LITERAL) {
+
+      // Format: event:target.method
+
+      // Event: "event:"
+      const event = tok.value;
```

Comment:
Redid parse a bit to deliver strings here instead of values. Added tests as well.